### DISCUSSION TITLE
one() get rows instead of row

### DISCUSSION THREAD

**Discussion by chlinwei** [sqlalchemy/sqlalchemy#12122] [Usage Questions] [answered]
```
class UserService:
    def getUserById(self,userId: int, session: SessionDep) -> UserDetailsSchema:
        stmt = (Select(UserModel).where(UserModel.id == userId))
        result = session.execute(stmt).one()
        print(len(result)) #1 
        return UserDetailsSchema.model_validate(result[0])
```

Hi,  I don't why the one() func return rows. instead of row .

**Comment by CaselIT** [comment]
Hi,

The normal result always returns tuple like objects. See https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Result.one for more details.

If tou want only a single element back you can use `scalars`. There are multiple options for that

**Reply by chlinwei** [reply, marked as answer]
Thank you for your prompt reply. I  got it .
